code/lib/cli/src/js-package-manager/Yarn2Proxy.ts

```diff
@@ -1,8 +1,65 @@
+import dedent from 'ts-dedent';
+import { createLogStream } from '../utils';
 import { JsPackageManager } from './JsPackageManager';
 import type { PackageJson } from './PackageJson';
 import type { InstallationMetadata, PackageMetadata } from './types';
 import { parsePackageData } from './util';
 
+const YARN2_ERROR_REGEX = /(YN\d{4}):.*?Error:\s+(.*)/i;
+const YARN2_ERROR_CODES = {
```

Comment:
Let's not curate this mapping manually. What is the reason that you do not just return the whole matching group of your regex findings? Can you elaborate on the benefits of manual mapping?

If we compare the different outputs:

1. Without mapping: `YN0001: │ Error: react@npm:28.2.0: No candidates found`
2. With mapping `YN0001 - EXCEPTION: react@npm:28.2.0: No candidates found`

I need help finding a clear benefit of the mapped message here.

Reply:
This change is not really for the user, but for us! The user will get the full logs in a log file and they can figure stuff by themselves.
For us, we get a crash report with that message, which means we can understand right away what it means, rather than us having to figure out what codes like `YN0024` or `EAI_FAIL` means.

The logic extracts a single line from the logs to provide the base of the error as part of the error message. For this simple example from the test, it might sound like it's not necessary, but some errors might only explain things further down the lines so I thought it'd be easier to be explicit on what the error means so we can get quick insights in our crash reports

Reply:
remove this!

Reply:
OK we actually decided to keep it to help us further on crash logs. I removed PNPM codes though